### DISCUSSION TITLE
How to find the corresponding commit based on the version number

### DISCUSSION THREAD

**Discussion by wenx1** [dotnet/runtime#58225] [Q&A] [answered]
When I was debugging a dotnet core console app, it shows the version of dotnet libraries is 4.700.21.21402.
How could I know which commit this version is linked to?

![image](https://user-images.githubusercontent.com/49379621/131052981-6d31169f-46ee-4a1b-b0d5-7473c970fdf5.png)

**Comment by gfoidl** [comment, marked as answer]
Look at that assemblies `AssemblyInformationalVersion` to get the commit hash.
E.g. by opening the assembly in tools like ILSpy.

**Reply by wenx1** [reply]
Thanks! It shows 3.1.15+585ae475d8aebdecf310d7509585003721509c5a. Which part of this string is the commit id?

**Reply by gfoidl** [reply]
The part ofter the `+` is git's commit hash / id.

**Reply by svick** [reply]
You can also see this in the Details tab of the Properties window:

![](https://user-images.githubusercontent.com/287848/131663182-726c750c-fd4c-49e3-abcf-1cfc913457b2.png)

You can also use GitHub's autolinking to get a link to the commit, as long as you know the right repo (and you don't even need the whole ID). E.g. for your version, you can use dotnet/corefx@585ae47. For 5.0.9+208e377a5329ad6eb1db5e5fb9d4590fa50beadd, that would be `dotnet/runtime@208e377`, which shows here as dotnet/runtime@208e377a5329ad6eb1db5e5fb9d4590fa50beadd.
